File: strategy/base/process.lua
--[[
--- Is run be the processor before / after each strategy run.
-- Subclass the class to create a new process
module "Process"
]]--
local Process = (require "../base/class").new("Process")

--- Execute the process actions here
-- @name run
function Process:run()
	error("stub")
end

--- Tells whether the process is finished.
-- Is called after each call to run
-- @name run
-- @return bool - Process is removed if true
function Process:isFinished()
	error("stub")
end

return Process

File: strategy/typescript/base/process.ts
/**
 * @module process
 * Is run by the processor before/after each strategy run.
 * Subclass the class to create a new process.
 */

/**************************************************************************
*   Copyright 2015 Alexander Danzer, Michael Eischer, Christian Lobmeier  *
*   Robotics Erlangen e.V.                                                *
*   http://www.robotics-erlangen.de/                                      *
*   info@robotics-erlangen.de                                             *
*                                                                         *
*   This program is free software: you can redistribute it and/or modify  *
*   it under the terms of the GNU General Public License as published by  *
*   the Free Software Foundation, either version 3 of the License, or     *
*   any later version.                                                    *
*                                                                         *
*   This program is distributed in the hope that it will be useful,       *
*   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
*   GNU General Public License for more details.                          *
*                                                                         *
*   You should have received a copy of the GNU General Public License     *
*   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
**************************************************************************/

export interface Process {
	/** Execute the process actions here */
	run(): void;

	/**
	 * Tells whether the process is finished.
	 * Is called after each call to run
	 * @returns Process is removed if true
	 */
	isFinished(): boolean;
}

